File: VB/WindowsFormsApplication2/PanelHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking
Imports DevExpress.XtraEditors

Namespace WindowsFormsApplication2
	Public Class PanelHelper
		Private manager As DockManager

		Public Sub New(ByVal manager As DockManager)
			Me.manager = manager
			AddHandler manager.RegisterDockPanel, AddressOf OnRegisterDockPanel
		End Sub

		Private Sub OnRegisterDockPanel(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Docking.DockPanelEventArgs)
			e.Panel.AllowDrop = True
			AddHandler e.Panel.DragOver, AddressOf OnPanelDragOver
		End Sub

		Private Sub OnPanelDragOver(ByVal sender As Object, ByVal e As DragEventArgs)
			Dim pnl As DockPanel = TryCast(sender, DockPanel)
			Dim pt As Point = pnl.PointToClient(New Point(e.X, e.Y))
			Dim info As HitInfo = pnl.GetHitInfo(pt)
			If info.HitTest = HitTest.Tab Then
				e.Effect = DragDropEffects.All
				info.Tab.Show()
			End If
		End Sub

		Private Sub OnUnregisterDockPanel(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Docking.DockPanelEventArgs)
			RemoveHandler e.Panel.DragOver, AddressOf OnPanelDragOver
		End Sub
	End Class
End Namespace

File: VB/WindowsFormsApplication2/PanelHelper.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking;
using DevExpress.XtraEditors;

namespace WindowsFormsApplication2
{
    public class PanelHelper
    {
        DockManager manager;

        public PanelHelper(DockManager manager)
        {
            this.manager = manager;
            manager.RegisterDockPanel += OnRegisterDockPanel;
        }

        private void OnRegisterDockPanel(object sender, DevExpress.XtraBars.Docking.DockPanelEventArgs e)
        {
            e.Panel.AllowDrop = true;
            e.Panel.DragOver += OnPanelDragOver;
        }
    
        private void OnPanelDragOver(object sender, DragEventArgs e)
        {
            DockPanel pnl = sender as DockPanel;
            Point pt = pnl.PointToClient(new Point(e.X, e.Y));
            HitInfo info = pnl.GetHitInfo(pt);
            if (info.HitTest == HitTest.Tab)
            {
                e.Effect = DragDropEffects.All;
                info.Tab.Show();
            }
        }

        private void OnUnregisterDockPanel(object sender, DevExpress.XtraBars.Docking.DockPanelEventArgs e)
        {
            e.Panel.DragOver -= OnPanelDragOver;
        }
    }
}
